config/crd/bases/configuration.konghq.com_kongclusterplugins.yaml

```diff
@@ -58,10 +67,6 @@ spec:
                   namespace:
                     description: The namespace containing the secret
                     type: string
-                required:
-                - key
-                - name
-                - namespace
```

Comment:
can you help me to better understand this removal? this would appear to loosen validation in a way that I'm not sure makes sense and it seems like perhaps we need some new build flags on the `KongClusterPlugin` object which would ensure these required keys get added.

Reply:
I forget where the doc for this is, but IIRC fields are required by default if not marked otherwise. I have no optional fields in the underlying type here, and don't mark its fields optional, but probably should have inferred that the KongPlugin package annotation sets the default optional flag for all nested types (OAPI doesn't know about the underlying Go types)

4fc7c2f319e8f572e25f5239c0ff877035172380